### PR TITLE
Generic method to obtain the SHA1 hash of the agent's and manager's key

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#9565|

## Description

This PR implements a new method to obtain the hash of the key used for the ciphered communication between the agent and the manager. It is accesible for both daemons and the `agent_auth` tool.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
- [x] Source installation
- [x] Review logs syntax and correct language

### CODE REVIEW

File: src/headers/sec.h [hunk 1]
```diff
@@ -170,6 +171,15 @@ int OS_DeleteSocket(keystore * keys, int sock);
...
 int w_get_agent_net_protocol_from_keystore(keystore * keys, const char * agent_id);
 
+/**
+ * @brief Receives a keyentry structure and returns the SHA1 value of the agent's key.
+ *
+ * @param key_entry The agent's key structure
+ * @param output The variable where the hashed key will be stored
+ * @retval Returns OS_SUCCESS on success or OS_INVALID on error.
+ **/
+int w_auth_hash_key(keyentry *key_entry, os_sha1 output);
```
Comment (palaciosjeremias) [inline src/headers/sec.h:181]: **w_auth** is a prefix for methods inside **auth.c** module.
As **keys.c** doesn´t have its own prefix, the correct name of this method could be:
`w_get_key_hash()`

Comment (pereyra-m) [reply src/headers/sec.h]: You are right, thank you !

File: src/os_crypto/shared/keys.c [hunk 2]
```diff
@@ -673,6 +673,23 @@ int OS_DeleteSocket(keystore * keys, int sock) {
 int w_get_agent_net_protocol_from_keystore(keystore * keys, const char * agent_id) {
 
     const int key_id = OS_IsAllowedID(keys, agent_id);
-    
+
     return (key_id >= 0 ? keys->keyentries[key_id]->net_protocol : key_id);
 }
+
+int w_get_key_hash(keyentry *key_entry, os_sha1 output) {
```
Comment (DProvinciani) [inline src/os_crypto/shared/keys.c:680]: Please, apply an indentation for the function according to the indentation used in the rest of the file.

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Just a minor change request.

File: src/os_crypto/shared/keys.c (hunk 2 above)
Comment (pereyra-m) [reply src/os_crypto/shared/keys.c:680]: You are right, fixed !

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**